service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -17,6 +17,7 @@
 import com.epam.esm.util.PageRequestDTOHandler;
 import org.springframework.stereotype.Service;
 import java.util.List;
+import java.util.Optional;
```

Comment:
You don't need  `private final TagRepository tagRepository` field in this class at all. You already have TagRepository in super class.

Reply:
I did this in order to use specific methods of TagRepository in the service : tagRepository.getMostPopularTag(user.getId())

Reply:
ok